pkg/cloud/ovirt/machine/actuator.go

```diff
@@ -376,10 +376,65 @@ func (actuator *OvirtActuator) reconcileConditions(
 	return conditions
 }
 
+
+// validateMachine execute several validations, example: check if user provided in
+// the machineset UserDataSecret, CredentialsSecret etc. If error exists, it will
+// return apierrors.MachineError
 func (actuator *OvirtActuator) validateMachine(machine *machinev1.Machine, config *ovirtconfigv1.OvirtMachineProviderSpec) *apierrors.MachineError {
-	return nil
+
+	errmsg := "error creating oVirt instance:"
+
+	// UserDataSecret
+	if config.UserDataSecret == nil {
+		return apierrors.CreateMachine(
+			fmt.Sprintf("%s UserDataSecret must be provided!", errmsg))
+	} else {
+		if config.UserDataSecret.Name == "" {
+			return apierrors.CreateMachine(
+				fmt.Sprintf("%s UserDataSecret *Name* must be provided!", errmsg))
+		}
+	}
+
+	// CredentialsSecret
+	if config.CredentialsSecret == nil {
+		return apierrors.CreateMachine(
+			fmt.Sprintf("%s CredentialsSecret must be provided!", errmsg))
+	} else {
+		if config.CredentialsSecret.Name == "" {
+			return apierrors.CreateMachine(
+				fmt.Sprintf("%s CredentialsSecret *Name* must be provided!", errmsg))
+		}
+	}
+
+	// If InstanceTypeId is NOT SET users can set MEM and CPU,
+	// all memory and cpu variables will be ignored.
+	if len(config.InstanceTypeId) == 0 {
+		if config.MemoryMB == 0 {
+			return apierrors.CreateMachine(
+				fmt.Sprintf("%s MemoryMB must be specified!", errmsg))
+		}
+		if config.CPU == nil {
+			return apierrors.CreateMachine(
+				fmt.Sprintf("%s CPU must be specified!", errmsg))
+		}
+	}
+
+	// root disk of the node
+	if config.OSDisk == nil {
+		return apierrors.CreateMachine(
+			fmt.Sprintf("%s OS Disk (os_disk) must be specified!", errmsg))
+	}
+
+	// VM Type: desktop, server, high_performance
+	if len(config.VMType) == 0 {
```

Comment:
We need to check that the VM type is not empty and contains: desktop, server, high-performance

```
allowedVMType := map[string]int{"desktop":0, "server":0, "high_performance":0}
if _, ok := allowedVMType[config.VMType]; !ok {
.....error
}
```